### PR TITLE
[JBPM-9554] jbpm.business.calendar.properties property not working inOCP environment

### PR DESCRIPTION
**Pull request by elguardian**
Add the possibility to specify business calendar outside the class path

### CODE REVIEW

File: jbpm-flow/src/test/java/org/jbpm/process/core/timer/BusinessCalendarImplTest.java [hunk 1]
```diff
@@ -387,6 +389,26 @@ public void testCalculateMinutesPassingWeekend() {
...
+        File file = new File(url.toURI());
+
+        BusinessCalendarImpl businessCal = new BusinessCalendarImpl(file.getAbsolutePath());
+        int myCustomProperty = businessCal.getPropertyAsInt("MyCustomProperty", "3");
+
+        assertEquals(7, myCustomProperty);
+    }
+
+    @Test
+    public void testLoadBadInputStream() throws Exception {
```
Comment (afalhambra) [inline jbpm-flow/src/test/java/org/jbpm/process/core/timer/BusinessCalendarImplTest.java:404]: there is no checked exception thrown by this method, please remove.
```suggestion
    public void testLoadBadInputStream() {
```

**Review by afalhambra — CHANGES_REQUESTED** [review]
Overall looks good to me, some minor cosmetic changes required.

File: jbpm-flow/src/main/java/org/jbpm/process/core/timer/BusinessCalendarImpl.java [hunk 2]
```diff
@@ -103,30 +106,43 @@
...
+        try (InputStream in = getInputStreamProperties(propertiesLocation)){
+            businessCalendarConfiguration.load(in);
+        } catch (IOException e) {
+           logger.error("Error while loading properties for business calendar {}", propertiesLocation, e);
         }
+
         init();
+    }
+
+    public InputStream getInputStreamProperties(String propertiesLocation) throws IOException {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-flow/src/main/java/org/jbpm/process/core/timer/BusinessCalendarImpl.java:126]: done

[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-flow/src/main/java/org/jbpm/process/core/timer/BusinessCalendarImpl.java:126]: done

File: jbpm-flow/src/test/java/org/jbpm/process/core/timer/BusinessCalendarImplTest.java (hunk 1 above)
Comment (elguardian) [reply jbpm-flow/src/test/java/org/jbpm/process/core/timer/BusinessCalendarImplTest.java:404]: done

File: jbpm-flow/src/main/java/org/jbpm/process/core/timer/BusinessCalendarImpl.java [hunk 3]
```diff
@@ -16,8 +16,11 @@
 
 package org.jbpm.process.core.timer;
 
+import java.io.File;
+import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.file.Files;
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-flow/src/main/java/org/jbpm/process/core/timer/BusinessCalendarImpl.java:23]: removed

**Review by afalhambra — APPROVED** [review]
Great job @elguardian

**Pull request merged**